### PR TITLE
[chore] Refactor label handling in repository automations

### PR DESCRIPTION
**Pull request by evan-bradley**
**Description:**

- Fix component regex for labels
- Fix component type substitution
- Enforce 50-character limit
- Provide common script to get component list

### CODE REVIEW

File: .github/workflows/scripts/generate-component-labels.sh
Comment (mx-psi) [inline .github/workflows/scripts/generate-component-labels.sh:18]: I guess that's `testbed/mockdatareceivers/mockawsxrayreceiver`?

**Review by mx-psi — APPROVED** [review]

File: .github/workflows/scripts/generate-component-labels.sh
Comment (evan-bradley) [reply .github/workflows/scripts/generate-component-labels.sh:18]: Nearly. That one is only 44 characters long, so it makes it through. The one that broke it was `processor/resourcedetectionprocessor/internal/azure/`, but with the new path to label translation, it is shortened to `processor/resourcedetection/internal/azure/`, so they're all under 50 characters now.

I'll adjust the wording to not claim there is necessarily a component without a label.

**Review by TylerHelmuth — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

File: .github/workflows/scripts/generate-component-labels.sh
Comment (astencel-sumo) [inline .github/workflows/scripts/generate-component-labels.sh:48]: `TYPE` is undefined

```console
./.github/workflows/scripts/generate-component-labels.sh: line 48: TYPE: unbound variable
```

**Comment by astencel-sumo** [comment]
I like how this change makes the labels shorter by removing the prefix from the second part of the label name, e.g. `processor/attributesprocessor` becomes just `processor/attributes` - clean and readable.

Also this actually shortens the one label that used to violate the 50-character label length limit in GitHub - `processor/resourcedetectionprocessor/internal/azure` - into a less-than-50-chars version `processor/resourcedetection/internal/azure`. :tada: :clap:

File: .github/workflows/scripts/generate-component-labels.sh
Comment (evan-bradley) [reply .github/workflows/scripts/generate-component-labels.sh:49]: Great catch! I usually comment this line out during my testing, I've made sure the variable is defined now.

File: .github/workflows/scripts/generate-component-labels.sh
Comment (jpkrohling) [inline .github/workflows/scripts/generate-component-labels.sh:17]: Which components are affected by this today?

Comment (evan-bradley) [reply .github/workflows/scripts/generate-component-labels.sh:17]: On the `main` branch, `processor/resourcedetectionprocessor/internal/azure` is currently affected. The new label conversion shortens this to `processor/resourcedetection/internal/azure`, so for the moment everything will be under this limit.

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Would it then make sense to require modules to be within this limit? I believe this feature might be significant enough to warrant this.

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: My biggest concern is building automation around this and leaving exceptions out (>50 chars labels), causing them to be hidden and forgotten when doing wider changes.

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: The solution is to skip the automation, which I agree is a solution to the problem. But doing that means that we need to be aware that this happened, which is unlikely to happen (unless we add a mention about it in a comment?).

Comment (mx-psi) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Sorry, I understand your concern with this PR's approach, but I don't understand what you are suggesting we do instead. What do you mean with 'skip the automation'?

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: > I don't understand what you are suggesting we do instead

What I had in mind was to create another check that would be applied to the new component's PR. This way, we'd know in advance if such a component would cause problems with our label automation.

Comment (mx-psi) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Okay, that sounds reasonable to me, thanks for clarifying, I guess what remains unclear to me is what do we do with the existing example that would fail that check?

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: I thought we didn't have a case where this would fail. Note that the component type is removed from the component's name: `processor/resourcedetection/internal/azure` (vs. `.../resourcedetectionprocessor/...`), so that this is within the limit.

Comment (mx-psi) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Right, I see now what I was misunderstanding, thanks for bearing with me :smile: I thought you wanted to check if the **component name** was longer than 50 chars, which would trigger the issue with `processor/resourcedetectionprocessor/internal/azure` but you want to check if the **label name** is longer than 50 chars. That makes total sense to me

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Right, sorry, I should have been clearer: I think we should be warned on the initial PR for a component if its name would go beyond the 50-char limit for its label.

Comment (evan-bradley) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Thank you both for your thoughts on this. To summarize the discussion, we are okay with allowing code ownership of paths that cannot turned into labels, but would like to be notified when a component would not have a label. Is this correct, or would we like to still consider requiring a label for each code ownership entry?

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Are labels being used only to ping code owners? I think that's a great feature, but if that's pretty much it, I would still allow such components to exist. If we plan on having further automation around the labels, we should block offending components from being merged.

Comment (evan-bradley) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Currently labels are only used to ping code owners, yes. If this approach sounds good to you for now, I'll open a follow-up issue to determine how to proceed from here.

Comment (jpkrohling) [reply .github/workflows/scripts/generate-component-labels.sh:17]: Sounds good!

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (jpkrohling) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:26]: A couple of new components were merged in the past few days (promtail receiver, purefa receiver). I believe they both had entries on those three templates, but do you think it would be worth rebasing this PR before merging?

Comment (evan-bradley) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:26]: Good idea. I believe those changes were included when I rebased this morning, but I've rebased again for good measure.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
